src/expr/src/scalar/mod.rs

```diff
@@ -1557,6 +1574,64 @@ mod tests {
                 },
                 output: err(EvalError::DivisionByZero),
             },
+            TestCase{
+                input: MirScalarExpr::CallVariadic{
+                    func: VariadicFunc::ListIndex,
+                    exprs: vec![
+                        MirScalarExpr::CallVariadic{
+                            func: VariadicFunc::ListCreate{elem_type: ScalarType::Int64},
+                            exprs: vec![col(0), col(1), col(2)]
+                        },
+                        lit(1)
+                    ]
+                },
+                output: col(0)
+            },
+            TestCase{
+                input: MirScalarExpr::CallVariadic{
+                    func: VariadicFunc::ListIndex,
+                    exprs: vec![
+                        MirScalarExpr::CallVariadic{
+                            func: VariadicFunc::ListCreate{elem_type: ScalarType::Int64},
+                            exprs: vec![col(0), col(1), col(2)]
+                        },
+                        lit(-1)
+                    ]
+                },
+                output: MirScalarExpr::CallVariadic{
+                    func: VariadicFunc::ListIndex,
+                    exprs: vec![
+                        MirScalarExpr::CallVariadic{
+                            func: VariadicFunc::ListCreate{elem_type: ScalarType::Int64},
+                            exprs: vec![col(0), col(1), col(2)]
+                        },
+                        lit(-1)
+                    ]
+                }
+            },
+            TestCase{
+                input: MirScalarExpr::CallVariadic{
+                    func: VariadicFunc::ListIndex,
+                    exprs: vec![
+                        MirScalarExpr::CallVariadic{
+                            func: VariadicFunc::ListCreate{elem_type: ScalarType::Int64},
+                            exprs: vec![col(0), col(1), col(2)]
+                        },
+                        lit(10)
+                    ]
+                },
+                output: MirScalarExpr::CallVariadic{
+                    func: VariadicFunc::ListIndex,
+                    exprs: vec![
+                        MirScalarExpr::CallVariadic{
+                            func: VariadicFunc::ListCreate{elem_type: ScalarType::Int64},
+                            exprs: vec![col(0), col(1), col(2)]
+                        },
+                        lit(10)
```

Comment:
Besides a test for multiple layers of nested lists, add a test for:
* `index = 0`, 
* `index = <list length + 1>`

Also, try varying the length of lists created and the expressions involved.